maven-plugins/enforcer-maven-plugin/src/main/java/io/helidon/build/maven/enforcer/nativeimage/NativeImageRule.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2022 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.helidon.build.maven.enforcer.nativeimage;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.LinkedList;
+import java.util.List;
+
+import io.helidon.build.common.maven.MavenVersion;
+import io.helidon.build.maven.enforcer.EnforcerException;
+import io.helidon.build.maven.enforcer.RuleFailure;
+
+import static io.helidon.build.common.maven.MavenVersion.toMavenVersion;
+
+/**
+ * Native image version rule.
+ */
+public class NativeImageRule {
+
+    private final List<VersionConfig> versionConfigs;
+
+    /**
+     * Constructor.
+     *
+     * @param config native-image configuration
+     */
+    public NativeImageRule(NativeImageConfig config) {
+        this.versionConfigs = config.rules();
+    }
+
+    /**
+     * Check native-image version.
+     *
+     * @return list of errors
+     */
+    public List<RuleFailure> check() {
+        List<RuleFailure> failures = new LinkedList<>();
+        Process process = startProcess();
+        try (BufferedReader reader = new BufferedReader(new InputStreamReader(process.getInputStream()))) {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                if (line.startsWith("GraalVM")) {
+                    checkVersion(failures, line.split(" "));
+                }
+            }
+        } catch (IOException e) {
+            throw new EnforcerException("Fail to run native-image --version");
+        }
+        return failures;
+    }
+
+    private void checkVersion(List<RuleFailure> failures, String[] elements) {
+        if (elements.length < 1) {
+            return;
+        }
+        MavenVersion version = toMavenVersion(elements[1]);
+        for (VersionConfig rule : versionConfigs) {
+            rule.checkVersion(version, failures);
+        }
+    }
+
+    private Process startProcess() {
+        ProcessBuilder builder = new ProcessBuilder("native-image", "--version")
```

Comment:
This goal needs to work well with both our native image plugin and the official one.

E.g. `helidon:native-image` resolves the path to `native-image` in various ways, we need to support that. Not sure what the official plugin does but we also need to support it.

Reply:
I checked the official plugin and they look for native image executable in `PATH`, `GRAALVM_HOME` and `JAVA_HOME` like `helidon-maven-plugin`.